server/testdriver/core/src/main/java/org/infinispan/server/test/core/InfinispanServerDriver.java

```diff
@@ -41,6 +41,16 @@ public interface InfinispanServerDriver {
     */
    void start(String name);
 
+   /**
+    * Starts an additional server outside that isn't part of {@link InfinispanServerTestConfiguration#numServers()}
+    * number. This is useful to start servers at a later point.
+    * <p>
+    * This method can only be invoked after {@link #start(String)} has completed successfully
+    * @param expectedClusterSize The expected number of members in the cluster. Some implementations may not work if other nodes
+    *                            outside of this driver are clustered and the argument could be ignored.
+    */
+   void startAdditionalServer(int expectedClusterSize);
```

Comment:
Ultimately we will want to have something like `StatefulSetRollingUpgradeIT`, but instead of a stopping and restart each container in order, we're replacing the stopped container with a container using a more recent server version.

This method when combined with `#stop` would allow the old container to be removed from the cluster and a new one added, but it doesn't seem to allow different versions to be provisioned as this is all configured via system properties in the `#start` method.

Reply:
This method is used to start additional servers after the first. So in this case this is used after we start an initial server list of 1 to add up to N additional servers. The rolling upgrade part is all in the jbang script I have currently that is not part of this PR. This PR is just to test to make sure my enhancements aren't breaking the existing tests :)

Reply:
Gotcha! I would be curious to see your JBang script if you have it somewhere easy to link?